pkg/chaosdaemon/dns_server.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chaosdaemon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/bpm"
+	pb "github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+)
+
+const (
+	// DNSServerConfFile is the default config file for DNS server
+	DNSServerConfFile = "/etc/resolv.conf"
+)
+
+func (s *daemonServer) SetDNSServer(ctx context.Context,
+	req *pb.SetDNSServerRequest) (*empty.Empty, error) {
+	log.Info("SetDNSServer", "request", req)
+	pid, err := s.crClient.GetPidFromContainerID(ctx, req.ContainerId)
+	if err != nil {
+		log.Error(err, "GetPidFromContainerID")
+		return nil, err
+	}
+
+	if req.Enable {
+		// set dns server to the chaos dns server's address
+
+		if len(req.DnsServer) == 0 {
+			return &empty.Empty{}, fmt.Errorf("invalid set dns server request %v", req)
+		}
+
+		// backup the /etc/resolv.conf
+		cmd := bpm.DefaultProcessBuilder("sh", "-c", fmt.Sprintf("ls %s.chaos.bak || cp %s %s.chaos.bak", DNSServerConfFile, DNSServerConfFile, DNSServerConfFile)).
```

Comment:
If we need to inject DNS failure on two pods and these two pods are on the same pod, what will be happen?

Reply:
Oh, sorry, this has no effect, this is the configuration of the pod.

Reply:
do you mean inject two DNS failure on one pod?

Reply:
ok